scanpy/readwrite.py

```diff
@@ -339,15 +335,23 @@ def read_visium(
             lowres_image=filedir / Path('spatial/tissue_lowres_image.png'),
         )
 
-        # check if files exists
+        # check if files exists, continue if images are missing
         for f in files.values():
             if not f.exists():
-                raise ValueError(f"Could not find '{f}'")
+                if any(x in str(f) for x in ["hires_image", "lowres_image"]):
+                    logg.warning(
+                        f"You seem to be missing an image file.\n"
+                        f"Could not find '{f}'."
+                    )
+                else:
+                    raise ValueError(f"Could not find '{f}'")
 
-        adata.uns['images'] = dict(
-            hires=imread(str(files['hires_image'])),
-            lowres=imread(str(files['lowres_image'])),
-        )
+        adata.uns['images'] = dict()
+        try:
+            adata.uns['images']['hires'] = imread(str(files['hires_image']))
+            adata.uns['images']['lowres'] = imread(str(files['lowres_image']))
+        except:
```

Comment:
Don’t use plain `except:`, as it will also try to catch unrecoverable errors.

If you really want to catch everything, do `except Exception`.

Reply:
Fixed